retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaCallFactory.java

```diff
@@ -129,80 +135,55 @@ public Request request() {
                 throw new IOException(e);
-            } catch (ExecutionException e) {
+            } catch (CompletionException e) {
                 throw new IOException(e.getCause());
             }
         }
 
         @Override
-        public void enqueue(Callback responseCallback) {
-            future = createRequest();
-            future.whenComplete((response, throwable) -> {
-                if (throwable != null) {
-                    responseCallback.onFailure(this, new IOException(throwable.getMessage(), throwable));
-                } else {
-                    if (!canceled) {
-                        try {
-                            responseCallback.onResponse(this, convertResponse(response));
-                        } catch (IOException e) {
-                            responseCallback.onFailure(this, e);
-                        }
-                    } else {
-                        responseCallback.onFailure(this, new IOException("Canceled"));
-                    }
-                }
-            });
+        public void enqueue(Callback callback) {
+            createRequest();
+            httpResponse.subscribe(new ArmeriaCallSubscriber(this, callback, request));
         }
 
         @Override
         public void cancel() {
-            canceled = true;
-            if (future != null) {
-                future.cancel(true);
-            }
+            executionStateUpdater.set(this, ExecutionState.CANCELED);
         }
 
         @Override
         public synchronized boolean isExecuted() {
-            return executed;
+            return httpResponse != null;
         }
 
         @Override
         public boolean isCanceled() {
-            return canceled;
+            return executionState == ExecutionState.CANCELED;
         }
 
-        private Response convertResponse(AggregatedHttpMessage httpMessage) {
-            Response.Builder builder = new Response.Builder();
-            String contentType = httpMessage.headers().get(HttpHeaderNames.CONTENT_TYPE);
-            httpMessage.headers().forEach(header -> {
-                builder.addHeader(header.getKey().toString(), header.getValue());
-            });
-            builder.request(request);
-            builder.code(httpMessage.status().code());
-            builder.message(httpMessage.status().reasonPhrase());
-            builder.protocol(Protocol.HTTP_1_1);
-            builder.body(ResponseBody.create(
-                    Strings.isNullOrEmpty(contentType) ? null : MediaType.parse(contentType),
-                    httpMessage.content().array()));
-            return builder.build();
+        boolean tryFinish() {
+            return executionStateUpdater.compareAndSet(this,
+                                                       ExecutionState.IDLE,
+                                                       ExecutionState.FINISHED)
+                   || executionStateUpdater.compareAndSet(this,
```

Comment:
Checkstyle violation here. Moving `||` to the end of the previous line will fix the problem.